src/gcovr/__main__.py

```diff
@@ -209,7 +209,7 @@ def load_config(partial_options) -> Dict[str, Any]:
     return {}
 
 
-def main(args=None):
```

Comment:
You're right, we can drop support of python 3.8 and change this but this needs also an adaption in the test matrix.

Reply:
If we drop support for Python 3.8 we need to drop support for gcc-5 and gcc-6 or solve one of the following problems:
- Install python 3.9 on Ubuntu 18.04 from source.
- Install gcc on Ubuntu 20.04 from source.

But this will solve the problem only until support of 3.9 is dropped.

Reply:
I managed to install python fro source.

Reply:
The Union syntax comes with python 3.10 and not with 3.9.
```
 error: X | Y syntax for unions requires Python 3.10  [syntax]
```

Reply:
Oh you're right, I misread the Python 3.9 release notes, which mentions "type" and "union", but not as part of the same feature:

> ![2024-11-26-python-type-union](https://github.com/user-attachments/assets/89e696dc-c67b-4a64-befe-250357f95e6a)